persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/configuration/RocksDBExpirationConfiguration.java

```diff
@@ -0,0 +1,75 @@
+package org.infinispan.persistence.rocksdb.configuration;
+
+import static org.infinispan.persistence.rocksdb.configuration.Element.EXPIRATION;
+
+import org.infinispan.commons.configuration.ConfigurationInfo;
+import org.infinispan.commons.configuration.attributes.Attribute;
+import org.infinispan.commons.configuration.attributes.AttributeDefinition;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.commons.configuration.elements.DefaultElementDefinition;
+import org.infinispan.commons.configuration.elements.ElementDefinition;
+
+/**
+ * @since 10.0
+ */
+public class RocksDBExpirationConfiguration implements ConfigurationInfo {
+
+   final static AttributeDefinition<String> EXPIRED_LOCATION = AttributeDefinition.builder("path", "Infinispan-RocksDBStore/expired").immutable().autoPersist(false).xmlName("path").build();
+   final static AttributeDefinition<Integer> EXPIRY_QUEUE_SIZE = AttributeDefinition.builder("queueSize", 10000).immutable().autoPersist(false).build();
```

Comment:
IMO these two attributes do not belong together, they should be regular attributes.

If anything, I would change the RocksDB configuration to have an element `<location base="Infinispan-RocksDBStore" data-prefix="data" expiration-prefix="expired"/>`, because every time you set the data location you should also set the expired location. Another option would be to remove the default value from the expired location attribute and programmatically set it to `data_location + "-expired"` when not set. Even better would be to use a [column family](https://github.com/facebook/rocksdb/wiki/Column-Families) for the expired information instead of a separate database...

Reply:
Could you update https://issues.jboss.org/browse/ISPN-10545 with your suggestion?